### PR TITLE
[Feature] create and implement warehouse crd

### PR DESCRIPTION
**Pull request by yandongxiao**

### CODE REVIEW

File: helm-charts/charts/warehouse/templates/cnconfigmap.yaml
Comment (kevincai) [inline helm-charts/charts/warehouse/templates/cnconfigmap.yaml:1]: rename this `cnconfigmap.yaml` , give a proper name?

File: helm-charts/charts/warehouse/templates/starrockswarehouse.yaml [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        {{- if .Values.spec.service.annotations }}
+        {{- toYaml .Values.spec.service.annotations | nindent 8 }}
+        {{- end }}
+      {{- end }}
+    {{- end }}
+    annotations:
+      app.starrocks.io/cn-config-hash: {{template "starrockswarehouse.config.hash" . }}
+      {{- if .Values.datadog.log.enabled }}
+      {{- if eq (trimAll " {}" .Values.datadog.log.logConfig) "" }}
+      ad.datadoghq.com/cn.logs: '[{"service":"starrocks","source":"cn"}]'
```
Comment (kevincai) [inline helm-charts/charts/warehouse/templates/starrockswarehouse.yaml:86]: may need attach the warehouse name to the tag as well.

File: helm-charts/charts/warehouse/values.yaml [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+    # besides the attributes you added, chart will append "source" and "service" attributes to the log config.
+    # see https://docs.datadoghq.com/containers/kubernetes/log/?tab=operator for more details.
+    logConfig: '{}' # e.g. '{"app": "starrocks", "tags": ["aa", "bb"]}'
+  metrics:
+    enabled: false
+
+spec:
+  # The name of starrocks cluster.
+  # Note: the starrocks cluster must be created before the warehouse by operator.
+  starRocksClusterName:
```
Comment (kevincai) [inline helm-charts/charts/warehouse/values.yaml:23]: may need to specify this field must not be empty?

File: helm-charts/charts/warehouse/templates/starrockswarehouse.yaml (hunk 1 above)
Comment (yandongxiao) [reply helm-charts/charts/warehouse/templates/starrockswarehouse.yaml]: done

File: helm-charts/charts/warehouse/values.yaml (hunk 2 above)
Comment (yandongxiao) [reply helm-charts/charts/warehouse/values.yaml:23]: done

**Review by kevincai — APPROVED** [review]

**Comment by kevincai** [comment]
the hardest part is to test this warehouse deployment. :)

**Pull request merged**